main.go

```diff
@@ -78,26 +79,52 @@ func main() {
 	srv := makeHandler(&config, NewDaemon(state))
 	http.Handle("/", srv)
 
-	if config.Insecure {
-		if config.TLSCert != "" {
-			log.Fatal("Error: Do not specify TLS certificate file",
-				" when Insecure is true.")
-		}
-		if config.TLSKey != "" {
-			log.Fatal("Error: Do not specify TLS key file",
-				" when Insecure is true.")
-		}
-		chkfatal(http.ListenAndServe(config.ListenAddr, nil))
-	} else {
-		if config.TLSCert == "" {
-			log.Fatal("Error: No TLS certificate file specified.")
-		}
-		if config.TLSKey == "" {
-			log.Fatal("Error: No TLS key file specified.")
-		}
+	host, _, err := net.SplitHostPort(config.ListenAddr)
+	if err != nil {
+		log.Fatal("Error parsing ListenAddr:", err)
+	}
+
+	hostIP := net.ParseIP(host)
+	// if host was a hostname rather than an ip address, hostIP will be nil,
+	// so this will correctly evaulate to false:
```

Comment:
Do you think it might cause confusion if someone were to write `localhost` thinking they are using a loopback address, but then obmd will complain that they are not using a loopback address? It might make sense to put it the message below that typing localhost will not work, we must specify by the ip.

Reply:
Good call, I'll make that explicit in the error message.

Reply:
Done.